<a href="https://colab.research.google.com/github/InnaTik/python_for_ds_task/blob/main/%D0%94%D0%97_2_%22hw_1_1_3_%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D0%B8_%D1%80%D0%BE%D0%B1%D0%BE%D1%82%D0%B8_%D0%B7_%D1%80%D1%8F%D0%B4%D0%BA%D0%B0%D0%BC%D0%B8_%D1%82%D0%B0_%D0%BC%D0%B0%D1%82%D0%B5%D0%BC%D0%B0%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D0%BE%D0%BF%D0%B5%D1%80%D0%B0%D1%86%D1%96%D1%97_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a=101
b=210
c=111
d=402

In [2]:
((a+b)*c)/d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [48]:
f=11283
d=89

e = f % d

print("Залишок від ділення:",e)

Залишок від ділення: 69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [7]:
f=11283
d=89

h = f // d

print("Ціла частина від ділення:",h)

Ціла частина від ділення: 126


4. Запишіть у змінну `name` ваше імʼя.

In [8]:
name = 'Inna'

5. Запишіть у змінну `surname` ваше прізвище.

In [9]:
surname = 'Tykhomyrova'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [10]:
name_surname = name + " " + surname

In [11]:
name_surname

'Inna Tykhomyrova'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [12]:
name_surname.upper()

'INNA TYKHOMYROVA'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [15]:
name_surname[2]

'n'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [23]:
print("Перша літера:", surname[0], "\nОстання літера:", surname[len(surname)-1])

Перша літера: T 
Остання літера: a


In [24]:
print("Перша літера:", surname[0], "\nОстання літера:", surname[-1])

Перша літера: T 
Остання літера: a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [25]:
len(surname)

11

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [27]:
city='Antananarivo'
city_new = city.lower()
city_new.count('a')

4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [38]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'

a = my_string.startswith('The')
b = my_string.endswith('dog!')

In [39]:
print("Починається рядок на 'The':", a,
      "\nЗакінчується рядок на 'dog!':", b)

Починається рядок на 'The': True 
Закінчується рядок на 'dog!': False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [47]:
a = my_string.split()
b = "_".join(a)

print(b)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`